itests/testdata/transfer_tx_testdata.go

```diff
@@ -49,38 +71,389 @@ type CommonTransferData struct {
+			nil,
+			utl.MinTxFeeWaves,
+			0,
+			utl.GetCurrentTimestampInMs(),
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "non-positive amount: 0",
+			},
+		),
+		"Asset amount > max": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
+			&assetId,
+			nil,
+			utl.MinTxFeeWaves,
+			utl.MaxAmount+1,
+			utl.GetCurrentTimestampInMs(),
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "failed to parse json message",
+			},
+		),
+		"Timestamp more than 7200000ms in the past relative to previous block timestamp": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
+			&assetId,
+			nil,
+			utl.MinTxFeeWaves,
+			uint64(assetAmount/4),
+			utl.GetCurrentTimestampInMs()-7260000,
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "is more than 7200000ms in the past relative to previous block timestamp",
+			}),
+		"Timestamp more than 5400000ms in the future relative to previous block timestamp": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
+			&assetId,
+			nil,
+			utl.MinTxFeeWaves,
+			uint64(assetAmount/4),
+			utl.GetCurrentTimestampInMs()+54160000,
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "is more than 5400000ms in the future relative to block timestamp",
+			}),
+		"Invalid asset ID": *NewTransferTestData(
+			utl.GetAccount(suite, utl.DefaultSenderNotMiner),
+			proto.NewRecipientFromAddress(utl.GetAccount(suite, utl.DefaultRecipientNotMiner).Address),
+			&invalidAssetId,
+			nil,
+			utl.MinTxFeeWaves,
+			uint64(assetAmount/4),
+			utl.GetCurrentTimestampInMs(),
+			utl.TestChainID,
+			nil,
+			TransferExpectedValuesNegative{
+				WavesDiffBalance:  0,
+				AssetDiffBalance:  0,
+				ErrGoMsg:          errMsg,
+				ErrScalaMsg:       errMsg,
+				ErrBrdCstGoMsg:    errBrdCstMsg,
+				ErrBrdCstScalaMsg: "Attempt to transfer unavailable funds: Transaction application leads to negative asset",
```

Comment:
should node make id validation before checking balance?

Reply:
This test was removed.